traffic_ops/client/servercapability.go

```diff
@@ -0,0 +1,100 @@
+/*
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package client
+
+import (
+	"encoding/json"
+	"fmt"
+	"net"
+	"net/http"
+
+	"github.com/apache/trafficcontrol/lib/go-tc"
+)
+
+const (
+	APIServerCapabilities = apiBase + "/server_capabilities"
+)
+
+// CreateServerCapability creates a server capability and returns the response
+func (to *Session) CreateServerCapability(sc tc.ServerCapability) (*tc.ServerCapabilityDetailResponse, ReqInf, error) {
+	var remoteAddr net.Addr
+	reqBody, err := json.Marshal(sc)
+	reqInf := ReqInf{CacheHitStatus: CacheHitStatusMiss, RemoteAddr: remoteAddr}
+	if err != nil {
+		return nil, reqInf, err
+	}
+	resp, remoteAddr, err := to.request(http.MethodPost, APIServerCapabilities, reqBody)
+	if err != nil {
+		return nil, reqInf, err
+	}
+	defer resp.Body.Close()
+	var scResp tc.ServerCapabilityDetailResponse
+	if err = json.NewDecoder(resp.Body).Decode(&scResp); err != nil {
+		return nil, reqInf, err
+	}
+	return &scResp, reqInf, nil
+}
+
+// GetServerCapabilities returns all the server capabilities
+func (to *Session) GetServerCapabilities() ([]tc.ServerCapability, ReqInf, error) {
+	resp, remoteAddr, err := to.request(http.MethodGet, APIServerCapabilities, nil)
+	reqInf := ReqInf{CacheHitStatus: CacheHitStatusMiss, RemoteAddr: remoteAddr}
+	if err != nil {
+		return nil, reqInf, err
+	}
+	defer resp.Body.Close()
+
+	var data tc.ServerCapabilitiesResponse
+	if err := json.NewDecoder(resp.Body).Decode(&data); err != nil {
+		return nil, reqInf, err
+	}
+
+	return data.Response, reqInf, nil
+}
+
+// GetServerCapability returns the given server capability by name
```

Comment:
Consider using `capabilities` to match the method signature. Also, is it ever expected to receive multiple capabilities? May be worth answering that question in the comment as well.

Reply:
Our API generally returns lists of resources even when it's only ever expected to return a single resource (as in this case). I suppose I could just have the client method return the single expected resource here, although most of our other client methods just return lists.